numba_cuda/numba/cuda/compiler.py

```diff
@@ -971,6 +972,173 @@ def define_error_gv(postfix):
     return helper_func
 
 
+def compile_all(
+    pyfunc,
+    sig,
+    debug=None,
+    lineinfo=False,
+    device=True,
+    fastmath=False,
+    cc=None,
+    opt=None,
+    abi="c",
+    abi_info=None,
+    output="ltoir",
+    forceinline=False,
+    launch_bounds=None,
+):
+    """Return a list of PTX/LTO-IR for kernel as well as external functions depended by the kernel.
+    If external functions are cuda c/c++ source, they will be compiled with NVRTC. The output code
+    kind is the same as the `output` parameter.
+    Otherwise, they will be passed through to the return list.
+    """
+
+    if output not in ("ptx", "ltoir"):
+        raise NotImplementedError(f"Unsupported output type: {output}")
+
+    if forceinline and output != "ltoir":
+        raise ValueError("Can only designate forced inlining in LTO-IR")
+
+    lto = output == "ltoir"
+
+    cc = _default_cc(cc)
+
+    lib, resty = _compile_pyfunc_with_fixup(
+        pyfunc,
+        sig,
+        debug=debug,
+        lineinfo=lineinfo,
+        device=device,
+        fastmath=fastmath,
+        cc=cc,
+        opt=opt,
+        abi=abi,
+        abi_info=abi_info,
+        forceinline=forceinline,
+        launch_bounds=launch_bounds,
+        lto=lto,
+    )
+
+    if lto:
+        code = lib.get_ltoir(cc=cc)
+    else:
+        code = lib.get_asm_str(cc=cc)
+    codes = [code]
+
+    # linking_files
+    lto = output == "ltoir"
```

Comment:
Nit: I'd consider renaming `lto` to `do_lto` or `is_lto` to more clearly indicate that its a flag controlling behaviour.

Reply:
The name here means whether the output should be ltoir, so `is_ltoir` is the best fit here IMO.